addons/website_slides/models/slide_slide.py

```diff
@@ -562,3 +587,56 @@ def _default_website_meta(self):
         res['default_opengraph']['og:description'] = res['default_twitter']['twitter:description'] = self.description
         res['default_opengraph']['og:image'] = res['default_twitter']['twitter:image'] = "/web/image/slide.slide/%s/image_thumb" % (self.id)
         return res
+
+    def get_user_relation(self, user):
+        relation = self.user_relation_ids.search([('user_id', '=', user)])
+        if relation:
+            return relation.state
+
+    @api.model
+    def reorder_slides(self, slides_data=None):
```

Comment:
Maybe it would be better to get an ordered list of ids and use `enumerate`, like a special controller did in 8.0 https://github.com/odoo/odoo/blob/8.0/addons/web/controllers/main.py#L976 ;)
Probably put this in a controller and avoid sudo. Who can resequence the lessons ?

Reply:
To my understanding, only the channel.user_id can resequence the slides